src/main/kotlin/interactor/Consumption.kt

```diff
@@ -5,58 +5,39 @@ import model.CityEntity
 
 class Consumption (private val dataSource: CostOfLivingDataSource){
 
-    fun execute(): String {
-         val x:MutableMap<Float,MutableList<String>> = mutableMapOf(Pair((0).toFloat(), mutableListOf()))
-         val cities=dataSource
+    fun execute(): CityEntity {
+         val x: MutableMap<Float,MutableList<CityEntity>> = mutableMapOf()
+         this.dataSource
             .getAllCitiesData()
-            .filter(::excludeNullValue)
-            .sortedBy{it.realEstatesPrices.apartment3BedroomsOutsideOfCentre}
-        cities.forEach {
-             val servicesCoast= servicesCoast(it)
-             val remainder= remainder(it,servicesCoast)
-             if(isSuitable(servicesCoast,remainder)){
-                 if (x.contains(remainder)) x[remainder]?.add(it.cityName)
-                 else x[remainder] = mutableListOf(it.cityName)
+            .filter(::excludeNullOrZeroValue)
+            .forEach {
```

Comment:
line 13 to the function return in line 20
try to make the code more cleaner and readable think in another way you can do the same thing